experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/unboxed.spec.ts

```diff
@@ -211,6 +279,254 @@ describe("unboxed unit tests", () => {
 				assert.equal(unboxed.value, true);
 			});
 		});
+
+		describe("Value field", () => {
```

Comment:
for the sake of fields (and thus unboxedField tests), all leaf nodes behave identically. I think it would be fine to have a single "leaf" test for the case of monomorphic leaves.

To be more specific, the thing that we really care about and should be testing are that:
1. the correct cases of fields unwrap
2. when fields unwrap, the invoke unwrapping recursively (in the case of fields, they invoke unwrapping on the node union below them)

Testing a single union of one leaf type (or anything else that does unwrapping) should be enough to cover that second part.

Separate tests for numbers and boolean and such is really just integration testing all three kinds of unwrapping a bunch of times, which doesn't seem valuable.